LinuxCommunicator/HostsFile/HostEntry.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Net;
+
+namespace Microsoft.Hpc.Communicators.LinuxCommunicator.HostsFile
+{
+    public class HostEntry
+    {
+        /// <summary>
+        /// The host name
+        /// </summary>
+        public string Name;
+
+        /// <summary>
+        /// The host IP address
+        /// </summary>
+        public string Address;
+
+        /// <summary>
+        /// Creates a new hosts file entry for the specified host.
+        /// </summary>
+        /// <param name="name">The host name</param>
+        /// <param name="ipAddress">The IP address</param>
+        public HostEntry(string name, string ipAddress)
+        {
+            IPAddress testAddress;
+            if (!IPAddress.TryParse(ipAddress, out testAddress))
```

Comment:
It's better not use exception as a condition. The TryXXX methods is designed for performance to avoid an exception catch, so converting the Try XXX to exceptions is something to be avoided.
I suggest you use bool HostEntry.TryCreate(name, ip);